Gordon360/Services/ProfileService.cs

```diff
@@ -282,6 +283,143 @@ public async Task UpdateProfileLinkAsync(string username, string type, CUSTOM_PR
+        public PublicFacultyStaffProfileViewModel ToPublicFacultyStaffProfileViewModel
+            (string username, string currentUserType, FacultyStaffProfileViewModel fac)
+        {
+            PublicFacultyStaffProfileViewModel publicFac = (PublicFacultyStaffProfileViewModel)fac;
+            var account = _accountService.GetAccountByUsername(username);
+
+            // select all privacy settings
+            var privacy = _context.UserPrivacy_Settings.Where(up_s => up_s.gordon_id == account.GordonID);
+            // get type of viewmodel for reflection property setting
+            Type fs_vm = new PublicFacultyStaffProfileViewModel().GetType();
+
+            foreach (UserPrivacy_Fields field in _context.UserPrivacy_Fields)
+            {
+                var row = privacy.FirstOrDefault(p => p.Field == field.Field);
+                if (row is UserPrivacy_Settings instance)
+                {
+                    if (instance.Visibility == "Private" || (instance.Visibility == "FacStaff" && currentUserType != "fac"))
+                    {
+                        fs_vm.GetProperty(field.Field).SetValue(publicFac, "Private as requested.");
+                    }
+                }
+                else
+                {
+                    fs_vm.GetProperty(field.Field).SetValue(publicFac, "Private as requested.");
+                }
+            }
+
+            return publicFac;
+        }
+
+        /// <summary>
+        /// convert original student profile to public student profile based on individual privacy settings
+        /// </summary>
+        /// <param name="username">username of the student being searched</param>
+        /// <param name="currentUserType">personnel type of the logged-in user</param>
+        /// <param name="stu">original profile of the student being searched</param>
+        /// <returns>public profile of the student based on individual privacy settings</returns>
+        public PublicStudentProfileViewModel ToPublicStudentProfileViewModel
+            (string username, string currentUserType, StudentProfileViewModel stu)
+        {
+            PublicStudentProfileViewModel publicStu = (PublicStudentProfileViewModel)stu;
+            var account = _accountService.GetAccountByUsername(username);
+
+            // select all privacy settings
+            var privacy = _context.UserPrivacy_Settings.Where(up_s => up_s.gordon_id == account.GordonID);
+            // get type of viewmodel for reflection property setting
+            Type s_vm = new PublicStudentProfileViewModel().GetType();
+
+            foreach (UserPrivacy_Fields field in _context.UserPrivacy_Fields)
+            {
+                var row = privacy.FirstOrDefault(p => p.Field == field.Field);
+                // HomePhone and SpouseName fields are not for students
+                if (field.Field == "HomePhone" || field.Field == "SpouseName") { }
```

Comment:
```suggestion
                // HomePhone and SpouseName fields are not for students
                if (field.Field == "HomePhone" || field.Field == "SpouseName") { }
```
Why are we doing a condition on the things that do not exist?

Reply:
This is because if I don't use the if statement here to due with the HomePhone and SpouseName field, current line 357 will throw an exception when it gets to field.Field = HomePhone or field.Field = SpouseName:
![image](https://github.com/gordon-cs/gordon-360-api/assets/78650688/4e914e00-22c4-4ca4-abbf-3965155d73c9)

(This is a bad explanation.. I can talk to you later in person if needed

Reply:
Yeah I think we can simplify this.